src/main/java/com/google/sps/servlets/RecipePostServlet.java

```diff
@@ -145,33 +149,36 @@ public void editRecipe(
     String title = request.getParameter("title");
     String imgURL = getUploadedFileUrl(request, "image");
     String description = request.getParameter("description");
-    int prepTime = Integer.parseInt(request.getParameter("time"));
+    Integer prepTime = Integer.parseInt(request.getParameter("time"));
     String difficulty = request.getParameter("difficulty");
     ArrayList<String> ingredients = getIngredients(request);
     ArrayList<String> stepList = getSteps(request);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
     Entity recipeEntity = null;
-
     try {
       recipeEntity = getRecipeById(datastore, request.getParameter("recipeId"));
     } catch (EntityNotFoundException e) {
       request.setAttribute("error", 1);
     }
     if (recipeEntity != null) {
-      recipeEntity =
-        setRecipeEntityProperties(
-          recipeEntity,
-          title,
-          imgURL,
-          description,
-          prepTime,
-          difficulty,
-          ingredients,
-          stepList,
-          request
-        );
+      Recipe recipe = new Recipe(
+        recipeEntity.getKey().getId(),
+        title,
+        description,
+        difficulty,
+        prepTime,
+        (String) request.getSession().getAttribute("name"),
+        imgURL,
+        stepList,
+        ingredients
+      );
+      recipeEntity = recipe.getEntity();
+      recipeEntity.setProperty(
+        "author_id",
+        request.getSession().getAttribute("id")
+      );
```

Comment:
You could have this property inside the recipe object directly, this way when you do a getEntity the author would be contained.